File: .idea/app/src/main/java/com/chevstrap/rbx/UI/ViewModels/Settings/ChevstrapViewModel.java
package com.chevstrap.rbx.UI.ViewModels.Settings;

import com.chevstrap.rbx.App;
import com.chevstrap.rbx.ConfigManager;
import com.chevstrap.rbx.Enums.RobloxAppType;
import com.chevstrap.rbx.Enums.ThemeRecreated;

import java.util.Map;
import java.util.Objects;

public class ChevstrapViewModel {
    public ChevstrapViewModel() {}

    public String getapp_theme_in_app() {
        String value = App.getChevstrapSettings().getAppThemeInApp();
        for (Map.Entry<ThemeRecreated, String> entry : ConfigManager.ThemeRecreateds.entrySet()) {
            if (Objects.equals(entry.getValue(), value)) {
                return entry.getKey().name();
            }
        }
        return ConfigManager.ThemeRecreateds.keySet().iterator().next().name();
    }

    public void setapp_theme_in_app(String modeName) {
        if (modeName == null) {
            return;
        }

        try {
            ThemeRecreated mode = ThemeRecreated.valueOf(modeName);
            App.getChevstrapSettings().setAppThemeInApp(ConfigManager.ThemeRecreateds.get(mode));
        } catch (IllegalArgumentException e) {
            App.getChevstrapSettings().setAppThemeInApp("dark");
        }
    }

    public boolean isForcePotraitMode() {
        return App.getChevstrapSettings().isForcePotraitMode();
    }

    public void setForcePotraitMode(boolean value) {
        App.getChevstrapSettings().setForcePotraitMode(value);
    }

    public String getTheLocale() {
        try {
            return App.getChevstrapSettings().getLocale();
        } catch (Exception e) {
            return null;
        }
    }

    public void setTheLocale(String value) {
        App.getChevstrapSettings().setLocale(value);
    }
}

File: app/src/main/java/com/chevstrap/rbx/ui/viewModels/settings/pages/ChevstrapViewModel.kt
package com.chevstrap.rbx.ui.viewModels.settings.pages

import com.chevstrap.rbx.App
import com.chevstrap.rbx.ConfigManager
import com.chevstrap.rbx.R
import com.chevstrap.rbx.SettingsActivity
import com.chevstrap.rbx.enums.ThemeRecreated
import com.chevstrap.rbx.extensions.ResourceManagerEx
import com.chevstrap.rbx.ui.Frontend
import com.chevstrap.rbx.ui.views.customDialogs.LogExplorerFragment
import kotlin.collections.iterator

class ChevstrapViewModel {

    var appThemeInApp: String
        get() {
            val value = App.config.data.appThemeInApp

            for ((mode, theme) in ConfigManager.ThemeRecreates) {
                if (theme == value) {
                    return mode.name
                }
            }

            return ConfigManager.ThemeRecreates.keys
                .firstOrNull()
                ?.name
                ?: ThemeRecreated.Dark.name
        }

        set(modeName) {
            try {
                val mode = ThemeRecreated.valueOf(modeName)
                App.config.data.appThemeInApp =
                    ConfigManager.ThemeRecreates[mode] ?: "dark"

                App.config.checkForChanges()
            } catch (_: IllegalArgumentException) {
                App.config.data.appThemeInApp = "dark"
            }
        }

    fun importCustomAppBackground() {
        val settingsActivity = App.savedSettingsActivity

        if (settingsActivity is SettingsActivity) {
            if (App.config.data.backgroundImageUri.isNotEmpty()) {
                Frontend.showMessageBoxWithRunnable(
                    App.savedFragmentActivity,
                    ResourceManagerEx.getStringOrEmpty(App.appContext, R.string.dialog_remove_custom_background),
                    true,
                    {
                        App.config.data.backgroundImageUri = ""
                        settingsActivity.removeBackgroundImage()
                    },
                    {}
                )
            } else {
                settingsActivity.imagePickerLauncher.launch("image/*")
            }
        }
    }

    fun logExplorer() {
        val settingsActivity = App.savedSettingsActivity

        if (settingsActivity is SettingsActivity) {
            LogExplorerFragment().show(
                settingsActivity.supportFragmentManager,
                "LogExplorerDialog"
            )
        }
    }

}